core/src/components/menu/menu.tsx

```diff
@@ -731,13 +717,26 @@ export class Menu implements ComponentInterface, MenuI {
       }
 
       // emit close event
-      this.ionDidClose.emit();
+      this.ionDidClose.emit({ role });
 
       // undo focus trapping so multiple menus don't collide
       document.removeEventListener('focus', this.handleFocus, true);
     }
   }
 
+  private onBackdropTap = (ev: any) => {
```

Comment:
I’m not sure if there are any consequences to moving this from the `click` listener, but all the other overlays use `onIonBackdropTap`.

Reply:
I was able to revert this change: https://github.com/ionic-team/ionic-framework/pull/29954/commits/f5179d9b729c91762bae12c5aa7f722b39695be2

When I implemented this feature in a different way this change was necessary. However, since I now pass the `role` in the close method, it no longer needs to be called in `onIonBackdropTap`.